src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java

```diff
@@ -300,6 +304,45 @@ private static Object getRecentActivity(Request req, Response res) {
         return activityList;
     }
 
+    /**
+     * Resends the user confirmation email for a given user
+     */
+    private static ObjectNode resendEmailConfirmation(Request req, Response res) {
+        // verify if the request is legit. Should only come from an application administrator or the user for which the
+        // account belongs to.
+        Auth0UserProfile userProfile = req.attribute("user");
+        if (userProfile.canAdministerApplication() || userProfile.getUser_id().equals(req.params("id"))) {
+            // authorized. Create request to resend email verification
+            HttpPost resendEmailVerificationRequest = new HttpPost(
+                "https://" + AUTH0_DOMAIN + API_PATH + "/jobs/verification-email"
+            );
+            setHeaders(req, resendEmailVerificationRequest);
+            setRequestEntityUsingJson(
+                resendEmailVerificationRequest,
+                JsonUtil.objectMapper.createObjectNode()
+                    .put("user_id", userProfile.getUser_id())
+                    .put("client_id", AUTH0_CLIENT_ID)
+                    .toString(),
+                req
+            );
+
+            // Execute request. If a HTTP response other than 200 occurs, an error will be returned.
+            executeRequestAndGetResult(resendEmailVerificationRequest, req);
+
+            // Email verification successfully sent! Return successful response.
+            return JsonUtil.objectMapper.createObjectNode()
+                .put("emailSent", true);
+        } else {
+            logMessageAndHalt(
```

Comment:
Can this else block get a comment about conditions that might lead to an unauthorized response?

Reply:
This comment is out of date now.